bundle/jsky.app.ot/src/main/scala/jsky/app/ot/editor/seq/ItcPanel.scala

```diff
@@ -273,6 +152,54 @@ class ItcSpectroscopyPanel(val owner: EdIteratorFolder) extends ItcPanel {
   }
 }
 
+/** Panel holding some feedback, mainly in case something went wrong with the ITC calculations. */
+private class ItcFeedbackPanel(table: ItcTable) extends Label {
+
+  import ItcPanel._
+  import OtColor._
+
+  opaque              = true
+  foreground          = Color.DARK_GRAY
+  border              = BorderFactory.createEmptyBorder(2, 2, 2, 2)
+  horizontalAlignment = Alignment.Left
+
+  listenTo(table.selection)
+  reactions += {
+    case TableRowsSelected(_, _, false)  => update()
+  }
+
+  private def update(): Unit = {
+    table.selected.fold {
+      visible = false                 // no table row selected, don't show feedback panel
+    } {
+      feedback(_).fold {
+        visible = false               // no feedback for this row, don't show feedback panel
+      } { case (ico, msg, col) =>
+        icon        = ico             // feedback available, show it
+        text        = msg
+        background  = col
+        visible     = true
+      }
+    }
+    revalidate()
+  }
```

Comment:
I think you could say `table.selected.flatMap(feedback).fold ...` to remove the nesting here.